### PR TITLE
[lua] Refactor Additional Effects to resolve cyclomatic complexity

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Moves additional effects into global functions that can be called from elsewhere
Remove cyclomatic complexity call
Also fix Onyos knife

## Steps to test these changes

Test every single add effect type and see no error
This PR does NOT fix broken things that have no visuals!

### CODE REVIEW

**Review by TeoTwawki — APPROVED** [review]
❤️

File: scripts/globals/additional_effects.lua [hunk 1]
```diff
@@ -110,209 +111,326 @@ xi.additionalEffect.calcDamage = function(attacker, element, defender, damage)
...
+    then
+        return 0, 0, 0 -- NMs immune or roll failed so return out
+    else
+        msgID = xi.msg.basic.ADD_EFFECT_STATUS
+        msgParam = xi.effect.KO
+        defender:setHP(0)
+    end
+
     return subEffect, msgID, msgParam
 end
```
Comment (Frankie-hz) [inline scripts/globals/additional_effects.lua]: There are two of these, not sure if this is intended?

Comment (WinterSolstice8) [reply scripts/globals/additional_effects.lua]: Whoops, sure isn't

Comment (WinterSolstice8) [reply scripts/globals/additional_effects.lua]: fixed

**Pull request merged**